server/src/config.rs

```diff
@@ -75,6 +75,22 @@ pub struct Opts {
     #[clap(long, env = "ATOMIC_DATA_DIR")]
     pub data_dir: Option<PathBuf>,
 
+    /// bucket name from s3-compatible storage service
+    #[clap(long, env = "ATOMIC_S3_BUCKET")]
+    pub s3_bucket: Option<String>,
+
+    /// region for s3-compatible storage service, defaults to "us-east-1"
+    #[clap(long, env = "ATOMIC_S3_REGION")]
+    pub s3_region: Option<String>,
+
+    /// endpoint for s3-compatible storage service, defaults to "https://s3.amazonaws.com"
+    #[clap(long, env = "ATOMIC_S3_ENDPOINT")]
+    pub s3_endpoint: Option<String>,
+
+    /// path where s3 uploads will be stored
+    #[clap(long, env = "ATOMIC_S3_PATH")]
+    pub s3_path: Option<String>,
```

Comment:
Perhaps also add a token / authentication param. Currently it only works with public buckets, right?

Reply:
There are multiple ways to authenticate. Currently it accepts a credentials file stored at `~/.aws/credentials` just like the aws cli (I'm assuming there's a similar path on windows) or the standard env vars of `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`

Reply:
But I'm also happy to add the auth options that opendal provides explicitly. We can add the access_key/secret_access_key and the token auth options

Reply:
My hesitation in adding support for these explicitly is it does complicate the code some, as using key/secret auth is mutually exclusive of using token auth. But of course something that we can handle if you think it has value over and above the authentication methods above.

Reply:
> There are multiple ways to authenticate. Currently it accepts a credentials file stored at `~/.aws/credentials` just like the aws cli (I'm assuming there's a similar path on windows) or the standard env vars of `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`

Wanted to make sure you saw this. This is also talked about in the documentation I wrote. Do you feel this is insufficient? For example, in our production applications at work this is how we authenticate to the AWS sdk.

Reply:
More details on credential files [here](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-format) which we could add to the documentation. Which is to say, the bucket doesn't have to be public.

Reply:
If given all of this you would still like an explicit way to set authentication in the code, I certainly can do it.